src/test/java/com/google/sps/data/DatastoreClientTest.java

```diff
@@ -0,0 +1,175 @@
+package com.google.sps.data;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import org.junit.Assert;
+import org.junit.Test;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import org.junit.After;
+import org.junit.Before;
+import java.util.Date;
+import java.util.Optional;
+import com.google.sps.data.Attendee;
+import com.google.sps.data.AttendeeInterface;
+import com.google.sps.data.Session;
+import com.google.sps.data.SessionInterface;
+import com.google.sps.data.Instance;
+import com.google.sps.data.InstanceInterface;
+import com.google.sps.data.DatastoreClient;
+import com.google.sps.data.DatastoreClientInterface;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import java.util.ArrayList;
+import java.util.List;
+
+/* Class that test the methods in DatastoreClient. */
+public class DatastoreClientTest  {
+  private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private final DatastoreClientInterface datastoreClient = 
+    new DatastoreClient();
+  private final DatastoreService DatastoreService = 
+    DatastoreServiceFactory.getDatastoreService();
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testAttendeeIsInsertedToDatastore() {
+    // Tests that object is inserted to datastore.
+    AttendeeInterface attendee1 = new Attendee("12345", "Taniece", new Date());
+    datastoreClient.insertOrUpdateAttendee(attendee1);
+    Assert.assertEquals(DatastoreService.prepare(new Query
+        (EntityConstants.AttendeeEntity.TABLE_NAME))
+        .countEntities(), 1);
+    // Tests the attendees with the same screenName but different sessionId are
+    // inserted as seperate entries.
+    AttendeeInterface attendee2 = new Attendee("54321", "Taniece", new Date());
+    datastoreClient.insertOrUpdateAttendee(attendee2);
+    Assert.assertEquals(DatastoreService.prepare(new Query
+        (EntityConstants.AttendeeEntity.TABLE_NAME))
+        .countEntities(), 2);    
+    // Tests that getAttendee finds and returns attendee.
+    AttendeeInterface attendee3 =
+        datastoreClient.getAttendee("Taniece", "12345").get();
+    Assert.assertEquals(attendee1, attendee3);
+    Assert.assertNotEquals(attendee1,attendee2);
+    // Tests that existing attendees are updated after being modified.
+    datastoreClient.insertOrUpdateAttendee(attendee2);
+    Assert.assertEquals(DatastoreService.prepare(new Query
+        (EntityConstants.AttendeeEntity.TABLE_NAME))
+        .countEntities(), 2);
+  }
+
+  @Test
+  public void testSessionIsInsertedToDatastore() {
```

Comment:
How does this pass lol? Did you change toEntity in Attendee to create a key thats a combination of name and session id? I dont see that change.

Reply:
I changed the key to screenName + sessionId